rpc/src/request.rs

```diff
@@ -14,10 +14,20 @@ pub trait Request: Debug + DeserializeOwned + Serialize + Sized + Send {
 
     /// Serialize this request as JSON
     fn into_json(self) -> String {
-        serde_json::to_string_pretty(&Wrapper::new(self)).unwrap()
+        Wrapper::new(self).into_json()
     }
 }
 
+/// Simple JSON-RPC requests which correlate with a single response from the
+/// remote endpoint.
+///
+/// An example of a request which is not simple would be the event subscription
+/// request, which, on success, returns a [`Subscription`] and not just a
+/// simple, singular response.
+///
+/// [`Subscription`]: struct.Subscription.html
+pub trait SimpleRequest: Request {}
```

Comment:
This is the simplest way I could find to configure the type system in such a way that I could restrict access to the `Client::perform` method to only those kinds of requests that expect a single response (i.e. non-subscription-related requests), while still preserving the existing overall type architecture.